parity/modules.rs

```diff
@@ -51,27 +61,30 @@ pub type SyncModules = (Arc<SyncProvider>, Arc<ManageNetwork>, Arc<ChainNotify>)
 mod ipc_deps {
 	pub use ethsync::{SyncClient, NetworkManagerClient, ServiceConfiguration};
 	pub use ethcore::client::ChainNotifyClient;
-	pub use hypervisor::{SYNC_MODULE_ID, BootArgs};
+	pub use hypervisor::{SYNC_MODULE_ID, BootArgs, HYPERVISOR_IPC_URL};
 	pub use nanoipc::{GuardedSocket, NanoSocket, init_client};
 	pub use ipc::IpcSocket;
 	pub use ipc::binary::serialize;
 }
 
 #[cfg(feature="ipc")]
-pub fn hypervisor() -> Option<Hypervisor> {
-	Some(Hypervisor::new())
+pub fn hypervisor(base_path: &Path) -> Option<Hypervisor> {
+	Some(Hypervisor
+		::with_url(&service_urls::with_base(base_path.to_str().unwrap(), HYPERVISOR_IPC_URL))
+		.io_path(base_path.to_str().unwrap()))
```

Comment:
`io_path` takes an `&str` and `Cow<str>` implements `Deref<Target=str>` so it should work if you just slap a `&*` on the front